### ISSUE TITLE
WordLlama Vs. Model2Vec

### ISSUE THREAD

**Issue by loretoparisi** [MinishLab/model2vec#97]
In my understanding you asses that M2V performs better on average of WordLlama when saying

_WL256, introduced in the WordLlama package is included for comparison due to its similarities to Model2Vec. However, we believe it is heavily overfit to the MTEB dataset since it is trained on datasets used in MTEB itself._

So, as general suggestion, for STS and Clustering downstream tasks, M2V would bring to better results for a given SBERT embeddings model.

Can you confirm this behavior as per your tests?

Is it possible to reproduce this comparison on a given Dataset/ Dataloader?

Thank you in advance!

**Comment by Pringled** [comment]
Hi @loretoparisi,

Yes, we believe that it is unfortunately overfit on some of the MTEB datasets and therefore the performance is not representative. You can confirm this by looking at the [used training datasets](https://github.com/dleemiller/WordLlama/blob/main/dataset_loader.py). 

We have performed extensive experiments to confirm this ourselves. [Here](https://github.com/MinishLab/model2vec/tree/main/results#mteb-results) you can see that WordLlama performs well on the MTEB datasets, but performs much worse on the datasets we introduced that are not in MTEB (PEARL and WordSim). 

This is further confirmed in [our classification benchmark](https://github.com/MinishLab/model2vec/tree/main/results#classification-and-speed-benchmarks), where we compared various models for classification on datasets that are not in MTEB. Here again you can see that the M2V models perform much better.

If you want to reproduce our results, you can use [our evaluation code](https://github.com/MinishLab/model2vec?tab=readme-ov-file#evaluation).

However, there is no one size fits all solution, so I would suggest to try both and see what works best for your own tasks/datasets.

**Comment by loretoparisi** [comment]
Thank you very much for the details!
My use cases are clustering (dimension 384) for hnsw indexing and top-k search (distance is L2 in faiss in that specific case) and ordinary cosine similarity of short texts.
Typically I would use onnx to achieve 2-3x speedup and avoid accuracy loss, so I was interested to test M2V approach to both get weights size and speed-up improvements.

**Comment by Pringled** [comment]
For clustering I would use one of our models with larger vocabularies (either [M2V_base_glove_subword](https://huggingface.co/minishlab/M2V_base_glove_subword) or [M2V_base_glove](https://huggingface.co/minishlab/M2V_base_glove) as those tend to perform a fair bit better on clustering tasks. Note that these use `bge-base-en-v1.5` as the base model and the vocabulary from GloVe, which both generalize well usually. However, depending on your data, it could be worth experimenting with your own distilled models, for example using a vocabulary based on your own data.

**Comment by loretoparisi** [comment]
> For clustering I would use one of our models with larger vocabularies (either [M2V_base_glove_subword](https://huggingface.co/minishlab/M2V_base_glove_subword) or [M2V_base_glove](https://huggingface.co/minishlab/M2V_base_glove) as those tend to perform a fair bit better on clustering tasks. Note that these use `bge-base-en-v1.5` as the base model and the vocabulary from GloVe, which both generalize well usually. However, depending on your data, it could be worth experimenting with your own distilled models, for example using a vocabulary based on your own data.

Thank you, forgot to mention that for STS embeddings is multi-lingual, so normally I use sbert multi language like

[sentence-transformers/distiluse-base-multilingual-cased-v2](https://huggingface.co/sentence-transformers/distiluse-base-multilingual-cased-v2)

**Comment by Pringled** [comment]
You're welcome! In that case I would suggest using our [multilingual model](https://huggingface.co/minishlab/M2V_multilingual_output), which is distilled from LaBSE. You can also try distilling from the multilingual model you are currently using to see what works better for your usecase.

**Comment by loretoparisi** [comment]
> You're welcome! In that case I would suggest using our [multilingual model](https://huggingface.co/minishlab/M2V_multilingual_output), which is distilled from LaBSE. You can also try distilling from the multilingual model you are currently using to see what works better for your usecase.

Thanks that's a good suggestion, used LaBSE in some tests in fact! 
Alt relevant discussion here

https://x.com/loretoparisi/status/1847611959039951334?s=46&t=LFe0jyu2TCnvXZNEvnb71g

**Comment by Pringled** [comment]
Closing this for now, feel free to open another issue if you have more questions about comparisons to other methods!
